components/molecule/phoneValidation/src/index.js

```diff
@@ -0,0 +1,145 @@
+import {useEffect, useRef, useState} from 'react'
+
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+
+import AtomIcon, {
+  ATOM_ICON_COLORS,
+  ATOM_ICON_SIZES
+} from '@s-ui/react-atom-icon'
+import MoleculeDropdownOption from '@s-ui/react-molecule-dropdown-option'
+
+import {getFlagEmoji, phoneValidationType} from './settings.js'
+
+const BASE_CLASS = 'sui-MoleculePhoneValidation'
+
+export default function MoleculePhoneValidation({
+  dropdownIcon,
+  maxLength = 11,
+  phone,
+  prefixes = [],
+  setPhone,
+  type = phoneValidationType.DEFAULT
+}) {
+  const [showDropdown, setShowDropdown] = useState(false)
+  const [selectedPrefix, setSelectedPrefix] = useState(prefixes[0])
+  const modalRef = useRef(null)
+
+  const baseClass = cx(
+    {
+      [`splitted`]: type === phoneValidationType.SPLITTED
+    },
+    BASE_CLASS
+  )
+
+  useEffect(() => {
+    const handleClickOutside = event => {
+      if (modalRef.current && !modalRef.current.contains(event.target)) {
+        setShowDropdown(false)
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [baseClass])
+
+  useEffect(() => {
+    phone && setPhone(`${selectedPrefix.countryCode} ${phone.split(' ')[1]}`)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPrefix])
+
+  return (
+    <div className={baseClass}>
+      <div className={`${baseClass}-input`}>
+        <div
+          className={`${baseClass}-input-prefix`}
+          testId="molecule-phone-validation-input-prefix"
+          onClick={() => setShowDropdown(!showDropdown)}
+        >
+          <span className={`${baseClass}-input-prefix-flag`}>
+            {getFlagEmoji(selectedPrefix.value)}
+          </span>
+          <AtomIcon
+            size={ATOM_ICON_SIZES.medium}
+            color={ATOM_ICON_COLORS.currentColor}
+          >
+            {dropdownIcon}
+          </AtomIcon>
+        </div>
+        <div className={`${baseClass}-input-phoneContainer`}>
```

Comment:
Same here, instead of do it natively, why we are not used a SUI Input component?

Reply:
I've used a native input for those reasons:

1. It's just an input that doesn't require any validation nor any tool that we have in the input dependency.
2. I need a transparent input with no borders and it doesn't exist in the design.
3. Avoid including unnecessary packages for the bundle size.

Reply:
If you think it's really needed please let me know, that was just a thought and It might be wrong.

Reply:
Nevermind, gonna use s-ui input since @turolopezsanabria suggested a change that includes a validation that it has hehe

Reply:
For me, using sui-input would be more tested than creating a custom one as you did.